testdata/fuzz/invariant/target/TargetSelectors.t.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity >=0.8.0;
+
+import "ds-test/test.sol";
+
+struct FuzzSelector {
+    address addr;
+    bytes4[] selectors;
+}
+
+contract Hello {
+    bool public world = true;
+
+    function change() public {
+        world = true;
+    }
+    
+    function real_change() public {
+        world = false;
+    }
+}
+
+contract TargetSelectors is DSTest {
+    Hello hello;
+
+    function setUp() public {
+        hello = new Hello();
+    }
+
+    function targetSelectors() public returns (FuzzSelector[] memory) {
```

Comment:
Thoughts on having a second type of `targetSelectors` that returns a struct of:

```solidity
struct FuzzArtifact {
  address addr;
  string artifactName;
}
```

This way I can define e.g. `interface IERC20 { ... }` in my test file and tell the fuzzer that some address has the interface defined by `"IERC20"`. Since return params don't impact the selector, maybe we rename the two methods to something like:
- `function targetFunctionsBySelector() external returns (FuzzSelector[] memory);`
- `function targetFunctionsByArtifact() external returns (FuzzArtifact[] memory);`

And the union of those two return values is what gets used. I've found sometimes you need to enumerate all selectors from an interface which is tedious/error-prone if you forget some

Reply:
I like the idea of generally having more granular ways & better UX to direct the fuzzer

Reply:
Can always pull this into a separate issue/PR if it's a lot of work or we want to iterate more on UX. Will defer to @joshieDo on this one

Reply:
yeah, we definitely need more filter capabilities through artifacts instead, but i'd make it in a separate PR.

Reply:
reminder for the followup^